icu4c/source/tools/icuexportdata/icuexportdata.cpp

```diff
@@ -96,6 +166,108 @@ void dumpEnumeratedProperty(UProperty uproperty, FILE* f) {
     usrc_writeUCPTrie(f, shortPropName, utrie.getAlias(), UPRV_TARGET_SYNTAX_TOML);
 }
 
+void dumpScriptExtensionsArray(FILE* f) {
+    IcuToolErrorCode status("icuexportdata: dumpScriptExtensionsArray");
+
+    const char* scFullPropName = u_getPropertyName(UCHAR_SCRIPT, U_LONG_PROPERTY_NAME);
+    const char* scxShortPropName = u_getPropertyName(UCHAR_SCRIPT_EXTENSIONS, U_SHORT_PROPERTY_NAME);
+
+    // We want to use 16-bits for our exported trie of sc/scx data because we
+    // need 12 bits to match the 12 bits of data stored for sc/scx in the trie
+    // in the uprops.icu data file.
+    UCPTrieValueWidth scWidth = UCPTRIE_VALUE_BITS_16;
+
+    // Create a mutable UCPTrie builder populated with Script property values data.
+    const UCPMap* scInvMap = u_getIntPropertyMap(UCHAR_SCRIPT, status);
+    handleError(status, scFullPropName);
+    LocalUMutableCPTriePointer builder(umutablecptrie_fromUCPMap(scInvMap, status));
+    handleError(status, scFullPropName);
+
+    // The values for the output scx companion array
+    std::vector< std::set<uint16_t>> outputDedupVec;
+
+    // The sc/scx companion array is an array of arrays (of script codes)
+    fputs("[script_extensions.script_code_array]\n", f);
```

Comment:
Nit: please print the array header and names before you print anything else:

```cpp
    fputs("[[script_extensions]]\n", f);
    fprintf(f, "long_name = \"%s\"\n", scFullPropName);
    if (scxShortPropName) fprintf(f, "short_name = \"%s\"\n", scxShortPropName);
```

Reply:
Done.